src/frontend/handlers.go

```diff
@@ -355,6 +362,21 @@ func (fe *frontendServer) chooseAd(ctx context.Context, ctxKeys []string, log lo
 	return ads[rand.Intn(len(ads))]
 }
 
+// convertAllCurrencies converts the currency for all products into every other currency
+// The error is logged as a warning since it is not critical.
+func (fe *frontendServer) convertAllCurrencies(r *http.Request, products []*pb.Product, currencies []string) {
+	log := r.Context().Value(ctxKeyLog{}).(logrus.FieldLogger)
+	for _, p := range products {
+		for _, c := range currencies {
```

Comment:
extract this to a method so it's a bit more masked :)

Reply:
fixed!